src/index.ts

```diff
@@ -0,0 +1,477 @@
+   */
+  quotePlaceholder: string;
+
+  /**
+   * Placeholder text to display in the quote's caption input.
+   */
+  captionPlaceholder: string;
+
+  /**
+   * Default alignment for the quote.
+   */
+  defaultAlignment: "center" | "left";
+}
+
+/**
+ * @typedef {object} QuoteData
+ * @description Quote Tool`s input and output data
+ * @property {string} text - quote`s text
+ * @property {string} caption - quote`s caption
+ * @property {'center'|'left'} alignment - quote`s alignment
+ */
+export interface QuoteData {
+  /**
+   * The text of the quote.
+   */
+  text: string;
+
+  /**
+   * The caption for the quote.
+   */
+  caption: string;
+
+  /**
+   * The alignment of the quote.
+   */
+  alignment: "center" | "left";
+}
+
+/**
+ * @typedef {object} TunesMenuConfig
+ * @property {string} icon - menu item icon
+ * @property {string} label - menu item label
+ * @property {boolean} isActive - true if item should be in active state
+ * @property {boolean} closeOnActivate - if true tunes menu should close once any item is selected
+ * @property {() => void} onActivate - item activation callback
+ */
+interface TunesMenuConfig {
+  /**
+   * Icon for the menu item.
+   */
+  icon: string;
+
+  /**
+   * Label for the menu item.
+   */
+  label: string;
+
+  /**
+   * Indicates if the item should be in an active state.
+   */
+  isActive: boolean;
+
+  /**
+   * Determines if the tunes menu should close after an item is selected.
+   */
+  closeOnActivate: boolean;
+
+  /**
+   * Callback function to be called when the item is activated.
+   */
+  onActivate: () => void;
+}
+
+/**
+ * @typedef {object} QuoteParams
+ * @description Constructor params for the Quote tool, use to pass initial data and settings
+ * @property {QuoteData} data - Preload data for the quote.
+ * @property {QuoteConfig} config - The configuration for the quote.
+ * @property {API} api - The Editor.js API.
+ * @property {boolean} readOnly - Is quote is read-only.
+ */
+interface QuoteParams {
+  /**
+   * Initial data for the quote
+   */
+  data: QuoteData;
+  /**
+   * Quote tool configuration
+   */
+  config: QuoteConfig;
+  /**
+   * Editor.js API
+   */
+  api: API;
+  /**
+   * Is quote read-only.
+   */
+  readOnly: boolean;
+}
+
+/**
+ * @typedef {object} QuoteCSS
+ * @description CSS classes names
+ * @property {string} block - Editor.js CSS Class for block
+ * @property {string} wrapper - Quote CSS Class
+ */
+interface QuoteCSS {
+  /**
+   * Editor.js CSS Class for block
+   */
+  baseClass: string;
+  /**
+   * Quote CSS Class
+   */
+  wrapper: string;
+  /**
+   * Quote CSS Class
+   */
+  input: string;
+  /**
+   * Quote CSS Class
+   */
+  text: string;
+  /**
+   * Quote CSS Class
+   */
+  caption: string;
+}
+
+/**
+ * @class Quote
+ * @classdesc Quote Tool for Editor.js
+ * @property {QuoteData} data - Tool`s input and output data
+ * @propert {API} api - Editor.js API instance
+ */
+export default class Quote {
```

Comment:
it should implement `BlockTool` interface from `@editorjs/editorjs`

Reply:
done👍🏼